PythonVisualizations/AdvancedSorting.py

```diff
@@ -0,0 +1,570 @@
+        # in Array whose d'th digit is == j
+
+        # convert the counts into cumulative counts:
+        for j in range(1, len(counts)): counts[j] += counts[j - 1]
+        # counts[j] now contains the number of values
+        # in arr whose d'th digit is <= to j
+
+        # place the values of arr into the correct slot of the answer array
+
+        x = canvas.coords(self.list[length-1].display_shape)[0]
+        y0 = canvas.coords(self.list[length-1].display_shape)[1] + CELL_SIZE/2
+
+        curArrow = canvas.create_line(x, y0, x - CELL_SIZE, y0, arrow="first", fill="red")
+        curTxt = canvas.create_text(x-30, y0-10, text="cur",
+                                     font=('Helvetica', '15'))
+
+        self.placeHolderArray()
+
+        for i in range(length - 1, -1, -1):
+            temp = (a[i].val // tenPower) % 10
+            counts[temp] -= 1
+            ans[counts[temp]] = a[i]
+
+            canvas.move(self.list[i].display_shape, CELL_SIZE*3, (counts[temp]-i) * CELL_SIZE)
+            canvas.move(self.list[i].display_val, CELL_SIZE*3, (counts[temp]-i) * CELL_SIZE)
+
+            window.update()
+            time.sleep(self.speed(0.8))
+            canvas.move(curArrow, 0, - CELL_SIZE)
+            canvas.move(curTxt, 0, - CELL_SIZE)
+
+        canvas.delete(curArrow)
+        canvas.delete(curTxt)
+        window.update()
+        self.list = ans
+        return True
+
+    def radixSort(self):
+        global running
+        # get the y coordinates of the next level down
+        shapeX = canvas.coords(self.list[0].display_shape)[0]
+        valX = canvas.coords(self.list[0].display_val)[0]
+        canvas.config(width=1200, height=600)
+
+        for i in range(0, len(self.list)):
+            cur = self.list[i]
+            canvas.move(cur.display_shape, -(canvas.coords(cur.display_shape)[0] - shapeX), CELL_SIZE*i - 30)
+            canvas.move(cur.display_val, -(canvas.coords(cur.display_val)[0] - valX), CELL_SIZE*i - 30)
+
+            window.update()
+            time.sleep(self.speed(0.01))
+
+        i = 0
+        while self.countingSortOnDigit(i):
+            i += 1
+
+        canvas.config(width = WIDTH, height = HEIGHT)
+        self.display()
+    '''
+    def drawPartition(self, left, right):
+        bottom = self.canvas.create_line(self.ARRAY_X0 + self.CELL_WIDTH*left, self.ARRAY_Y0 + self.CELL_SIZE + 90,
+                           self.ARRAY_X0 + self.CELL_WIDTH*(right+1), self.ARRAY_Y0 + self.CELL_SIZE + 90, fill="red")
+        l = self.canvas.create_line(self.ARRAY_X0 + self.CELL_WIDTH*left, self.ARRAY_Y0 + self.CELL_SIZE + 80,
+                           self.ARRAY_X0 + self.CELL_WIDTH * left, self.ARRAY_Y0 + self.CELL_SIZE + 90, fill="red")
+        r = self.canvas.create_line(self.ARRAY_X0 + self.CELL_WIDTH * (right+1), self.ARRAY_Y0 + self.CELL_SIZE + 80,
+                           self.ARRAY_X0 + self.CELL_WIDTH * (right+1), self.ARRAY_Y0 + self.CELL_SIZE + 90, fill="red")

```

Comment:
Since the partition is based on the array cells and its best to avoid repeating the calculations of where the cell boundaries are, the `drawPartition()` method would be better defined by using `cellCoord()`.  You can also combine all three line segment items into a single line item to make the bracket.
```
        leftCoords = self.cellCoords(left)
        rightCoords = self.cellCoords(right)
        dy = 10
        bracket = self.canvas.create_line(
            leftCoords[0], leftCoords[3] + dy, 
            leftCoords[0], leftCoords[3] + 2 * dy, 
            rightCoords[2], rightCoords[3] + 2 * dy,
            rightCoords[2], rightCoords[3] + dy, fill = "red")
```